lib/wine_o_inventory_api_web/endpoint.ex

```diff
@@ -44,5 +44,6 @@ defmodule WineOInventoryApiWeb.Endpoint do
   plug Plug.MethodOverride
   plug Plug.Head
   plug Plug.Session, @session_options
+  plug CORSPlug, origin: "*"
```

Comment:
Why was this needed?

Reply:
While working on the frontend, my requests weren't able to hit the endpoint due to CORS not being enabled on the backend. Phoenix's CORSPlug allows me to enable cross resource sharing. I should be more specific and not use the wildcard for origin now that I think about it. Thoughts?

Reply:
Thanks for asking this question if forced me to refresh my CORS knowledge. CORS is essentially away to relax the [same origin policy](https://en.wikipedia.org/wiki/Same-origin_policy#Origin_determination_rules). The same origin policy determination is same scheme, host and port. So when your server is on http://localhost:3000 and your frontend is on http://localhost:8081 you have the same scheme, the same host, but not the same port, therefore you aren't able to access the localhost:3000 from localhost:8081 — they don't comply with the same origin policy. You can get around this with CORS, which is a relaxation of this policy. CORS allows you to specify, who or what can access your server's endpoint. As such, you need to be thoughtful about when you enable CORS. This article provides a pretty good overview of the security risks: https://nodeployfriday.com/posts/cors-cyber-attacks/